Sources/AWSLambdaRuntimeCore/LambdaHandler.swift

```diff
@@ -26,7 +26,7 @@ import NIOCore
 ///         level protocols ``EventLoopLambdaHandler`` and
 ///         ``ByteBufferLambdaHandler``.
 @available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
-public protocol LambdaHandler: EventLoopLambdaHandler {
+public protocol LambdaHandler: EventLoopLambdaHandler where Event: _AWSLambdaSendable {
```

Comment:
@fabianfett this is an interesting bit. since we are using it in an async context in L62 it needs to be sendable, or we can wrap it if we are sure access to it is safe

Reply:
The Event does not need to be `Sendable`, access will always be serial.

Reply:
Another question IMO is, should we require the Output to be Sendable? Users could do something like this:

```
struct MyHandlerThatTriesToDoSomethingBad {
  func handle(_ event: Event, context: LambdaContext) async throws -> MyOutput {
    var output = Foo()
    Task {
      output.bar() // <--- compiler will require Sendable here!
    }
    return output
  }
}
```

I don't think we need to require Output to be Sendable as well. As the example above shows, as soon as the user would do something illegal, the compiler would step in and require sendability, if it is needed.

Reply:
@fabianfett what if the users want to use the event payload in the task?

struct MyHandlerThatTriesToDoSomethingBad {
  func handle(_ event: Event, context: LambdaContext) async throws -> MyOutput {
    var output = Foo()
    Task {
        someFunc(event)
    }
    return output
  }
}

Reply:
@tomerd That is a valid point. For this reason we should make all the events we expose in `swift-aws-lambda-events` Sendable. But the *Runtime* does not and should not need to require it.